File: week2/output.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

// Function to read a file and just read its contents without printing them
void read_file(const char *file_path) {
    FILE *file = fopen(file_path, "r");
    if (!file) {
        printf("Could not open file %s for reading\n", file_path);
        return;
    }

    // Just read through the file without printing its contents
    char ch;
    while ((ch = fgetc(file)) != EOF) {
        // Do nothing with 'ch' to avoid printing, just reading
    }

    fclose(file);
}

int main() {
    const char *file_path = "data/output.txt";  // Change this path as needed

    clock_t start_time = clock();  // Start timer
    read_file(file_path);  // Read the file
    clock_t end_time = clock();  // End timer

    double time_taken = ((double)(end_time - start_time)) / CLOCKS_PER_SEC;  // Calculate time taken
    printf("\nTime taken to read the file: %.5f seconds\n", time_taken);

    return 0;
}

File: week2/output.py
import time

# Function to read a file and return its contents
def read_file(file_path):
    with open(file_path, 'r') as file:
        return file.read()

# Time how long it takes to read the file
def time_file_read(file_path):
    start_time = time.time()  # Start timer
    file_contents = read_file(file_path)  # Read the file
    end_time = time.time()  # End timer
    
    time_taken = end_time - start_time  # Calculate time taken
    print(f"Time taken to read the file: {time_taken} seconds")
    
    return file_contents  # Return contents in case you want to process it

# Example file path
file_path = 'data/output.txt'  # Change this path if needed

# Run the function and see how long it takes
contents = time_file_read(file_path)

# Optionally, print the contents if needed (for small files)
# print(contents)

